### PR TITLE
Add structured logs test for querystring parameters

### PR DESCRIPTION
**Pull request by JamesNK**
Test for regression fixed in https://github.com/dotnet/aspire/pull/5023

`ProtectedLocalStorage` was abstracted to make it injectable into the unit test. This caused a bit of churn, but the API and function remains the same.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5038)

### CODE REVIEW

**Review by drewnoakes — APPROVED** [review]

File: src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs [hunk 1]
```diff
@@ -24,7 +23,7 @@ public sealed partial class ConsoleLogs : ComponentBase, IAsyncDisposable, IPage
     public required IDashboardClient DashboardClient { get; init; }
 
     [Inject]
-    public required ProtectedSessionStorage SessionStorage { get; init; }
+    public required ILocalStorage SessionStorage { get; init; }
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs:26]: I didn't notice they were different. I'll fix.

File: src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor.cs [hunk 2]
```diff
@@ -62,20 +58,23 @@ public partial class StructuredLogs : IPageWithSessionAndUrlState<StructuredLogs
     public required BrowserTimeProvider TimeProvider { get; init; }
 
     [Inject]
-    public required ILogger<Traces> Logger { get; init; }
+    public required ILogger<StructuredLogs> Logger { get; init; }
```
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/StructuredLogs.razor.cs:61]: In this situation, the type is wrong, but there are situations where you would want the logger type to not match it's fields containing type.

File: src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (JamesNK) [reply src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs]: Unintended. Fixed.

**Pull request merged**